libs/librepcb/projecteditor/boardeditor/fsm/bes_addvia.cpp

```diff
@@ -330,22 +359,115 @@ bool BES_AddVia::updateVia(Board& board, const Point& pos) noexcept {
   }
 }
 
-bool BES_AddVia::fixVia(const Point& pos) noexcept {
-  Q_ASSERT(mUndoCmdActive == true);
+bool BES_AddVia::fixVia(Point& pos) noexcept {
+  Q_ASSERT(mSubState == SubState_PositioningVia);
+  //TODO(5n8ke): handle user errors in a more graceful way without popup message
 
   try {
     mViaEditCmd->setPosition(pos, false);
+
+    Board* board = &mCurrentVia->getBoard();
```

Comment:
That looks strange - if you get a reference to a board, you should store it as a reference since this way it's clear that it can never be null.

```suggestion
    Board& board = mCurrentVia->getBoard();
```

But actually I would even pass `Board& board` as a parameter to the `fixVia()` method (just like `updateVia()`) instead of fetching it from the currently placed via. The only function which calls `fixVia()` even already has a pointer to the board.

Reply:
Done.